### ISSUE TITLE
Upgrade to AzureRM 3.3.0

### ISSUE THREAD

**Issue by eyenx** [Azure/terraform-azurerm-aks#156]
AzureRM 3.0 has been out there for some time now. In my eyes it's time to upgrade this module to use AzureRM 3.3.0

The upgrade guide clearly states what needs to be changed:

https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/guides/3.0-upgrade-guide#resource-azurerm_kubernetes_cluster

**Comment by nikolay** [comment]
Today, v3.5.0 is the latest. I am starting with Azure today and there's already going to be a technical debt and upcoming upgrade since day 1, which is not a pleasant feeling!

**Comment by eyenx** [comment]
I am starting to get the feeling I would be better off by not using this poorly maintained module. It is way faster to just use the Terraform resources directly + you are up to date!

**Comment by nikolay** [comment]
I agree with you, @eyenx! I made the mistake so many times - I start using a Terraform module, which suddenly stops being maintained, and then I have to fork it due to licensing considerations and then use a small module for almost no reason. I was initially pleasantly surprised that Microsoft is offer modules for their cloud unlike Amazon, but it seems they are doing more harm than good!

**Comment by brunoscota** [comment]
Hey Microsoft if you wont mantain this please close this repo!

**Comment by davidkarlsen** [comment]
@lonegunmanb - do you work on this module too?

**Comment by lonegunmanb** [comment]
@davidkarlsen Yeah. I've talked with my team, since we haven't decided which module will be our first module to add ci pipeline, I'll work on this pr this week.
